tests/container_structure_report_generation_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib"
+	"github.com/GoogleContainerTools/container-structure-test/cmd"
+	types "github.com/GoogleContainerTools/container-structure-test/pkg/types/unversioned"
+	"github.com/spf13/cobra"
+)
+
+var TestImage = "testImage"
+var TestConfig = "dummy"
+var GenerateSuccessResult = true
+var TestChannel chan interface{}
+
+func addResults(b bool) {
+	result := &types.TestResult{
+		Name:   "TestResult Test",
+		Pass:   true,
+		Errors: make([]string, 0),
+	}
+	if !b {
+		result.Pass = false
+		result.Errors = append(result.Errors, "Test Error")
+	}
+	TestChannel <- result
+	close(TestChannel)
+}
+
+func streamOTest(command *cobra.Command, args []string) {
+	go addResults(GenerateSuccessResult)
+}
+
+func TestContainerStructureTestsTestReportGenerationWhenCmdSucceds(t *testing.T) {
+	TestChannel = make(chan interface{}, 1)
```

Comment:
There's a bit of boilerplate here that might be reducable, making the test itself more understandable. Maybe a table-driven test could help.